### PR TITLE
Issue #135  Improving logging within Proxy

### PR DESCRIPTION
**Pull request by hferentschik**
Also adding some helper scripts for local development

Addresses issue #135

### CODE REVIEW

File: internal/proxy/proxy.go [hunk 1]
```diff
@@ -328,35 +358,35 @@ func (p *Proxy) setIdledCookie(w http.ResponseWriter, pci ProxyCacheItem) {
...
 	}
+
 	err = json.Unmarshal(body, &gh)
 	if err != nil {
-		p.HandleError(w, err)
+		p.HandleError(w, err, requestLogEntry)
 		return
 	}
 
+	requestLogEntry.WithField("json", gh).Info("Processing GitHub JSON payload")
```
Comment (vpavlin) [inline internal/proxy/proxy.go]: I'd do this as `Debug` - probably no need to log all request payloads?

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -461,8 +491,22 @@ func (p *Proxy) processToken(tokenData []byte) (pci ProxyCacheItem, osioToken st
...
 //HandleError creates a JSON response with a given error and writes it to ResponseWriter
-func (p *Proxy) HandleError(w http.ResponseWriter, err error) {
-	log.Error(err)
+func (p *Proxy) HandleError(w http.ResponseWriter, err error, requestLogEntry *log.Entry) {
+	// log the error
+	location := ""
+	if err != nil {
+		pc, fn, line, _ := runtime.Caller(1)
+
+		location = fmt.Sprintf(" %s[%s:%d]", runtime.FuncForPC(pc).Name(), fn, line)
```
Comment (vpavlin) [inline internal/proxy/proxy.go]: This is awesome:)

File: internal/proxy/proxy.go [hunk 3]
```diff
@@ -476,60 +520,72 @@ func (p *Proxy) HandleError(w http.ResponseWriter, err error) {
...
 
 //GetUser returns a namespace name based on GitHub repository URL
-func (p *Proxy) GetUser(pl GHHookStruct) (namespace string, err error) {
+func (p *Proxy) GetUser(pl GHHookStruct, requestLogEntry *log.Entry) (string, error) {
 	if n, found := p.TenantCache.Get(pl.Repository.CloneURL); found {
-		log.Info("Cache hit")
-		namespace = n.(string)
-		return
+		namespace := n.(string)
+		requestLogEntry.Infof("Cache hit for repository %s: %s", pl.Repository.CloneURL, namespace)
```
Comment (vpavlin) [inline internal/proxy/proxy.go]: Maybe add `ns` field?

File: internal/proxy/proxy.go [hunk 4]
```diff
@@ -476,60 +520,72 @@ func (p *Proxy) HandleError(w http.ResponseWriter, err error) {
...
 }
 
 //RecordStatistics writes usage statistics to a database
 func (p *Proxy) RecordStatistics(ns string, la int64, lbf int64) (err error) {
-	log.WithField("ns", ns).Info("Recording stats")
+	log.WithField("ns", ns).Debug("Recording stats")
 	s, notFound, err := p.storageService.GetStatisticsUser(ns)
 	if err != nil {
-		log.Warningf("Could not load statistics for %s: %s (%v)", ns, err, s)
+		log.Warningf("Could not load statistics for namespace %s: %s", ns, err)
```
Comment (vpavlin) [inline internal/proxy/proxy.go]: Could you add `ns` field?

File: internal/proxy/proxy.go (hunk 1 above)
Comment (hferentschik) [reply internal/proxy/proxy.go]: Fair enough. I was undecided on this one. Let's do debug then.

File: internal/proxy/proxy.go (hunk 2 above)
Comment (hferentschik) [reply internal/proxy/proxy.go]: thanks :-) Definitely adds some more context.

File: internal/proxy/proxy.go (hunk 3 above)
Comment (hferentschik) [reply internal/proxy/proxy.go]: +1

Comment (hferentschik) [reply internal/proxy/proxy.go]: On that note, any update why Kibana is not splitting our JSON?

File: internal/proxy/proxy.go (hunk 4 above)
Comment (hferentschik) [reply internal/proxy/proxy.go]: +1

**Comment by hferentschik** [comment]
@vpavlin  Pushed update

File: scripts/setupLocalProxy.sh [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    forwardIdler &
+    forwardTenant &
+    runPostgres &
+
+    echo export JC_KEYCLOAK_URL=https://sso.prod-preview.openshift.io
+    echo export JC_WIT_API_URL=https://api.prod-preview.openshift.io
+    echo export JC_REDIRECT_URL=https://jenkins.prod-preview.openshift.io
+    echo export JC_AUTH_URL=https://auth.prod-preview.openshift.io
+    echo export JC_POSTGRES_PORT=${LOCAL_POSTGRES_PORT}:
```
Comment (vpavlin) [inline scripts/setupLocalProxy.sh]: Extra colon at the end

Comment (hferentschik) [reply scripts/setupLocalProxy.sh]: bugger, once sec

**Comment by vpavlin** [comment]
LGTM

**Pull request merged**